ouroboros-consensus/test-consensus/Test/Dynamic/Util/TestConfig.hs

```diff
@@ -0,0 +1,482 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving  #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+-- | Test configuration, including per-protocol config
```

Comment:
I'm fine with splitting this up in a module per protocol, even if that requires orphan instances. Up to you.

Reply:
I minimized this PR, so this content will be in a different PR. Resolving this conversation.